metrics/api/src/main/java/io/helidon/metrics/api/SystemTagsManager.java

```diff
@@ -119,6 +120,18 @@ static void onChange(Consumer<SystemTagsManager> changeListener) {
      */
     Iterable<Tag> displayTags();
 
+    /**
+     * Returns name/value pairs of system tags, avoiding constructing the provider's {@link io.helidon.metrics.api.Tag}
+     * implementations for each.
+     *
+     * @return system tag name/value pairs
+     */
+    default Map<String, String> displayTagPairs() {
+        Map<String, String> result = new TreeMap<>();
+        displayTags().forEach(tag -> result.put(tag.key(), tag.value()));
+        return result;
```

Comment:
Should this be unmodifiable?

Reply:
What would this being unmodifiable protect against? The returned `Map` does not use the original for backing.

Reply:
True enough. Although without the unmodifiable contract, then it is OK for any caller to do, for example, something stupid like `stm.displayTagPairs().clear()`. Obviously that's their problem, not yours, but it's part of the contract. (The alternative is documenting that it will throw an `UnsupportedOperationException` if a mutation is applied to the returned `Map`.)